geomstats/geometry/special_orthogonal.py

```diff
@@ -1,1664 +1,34 @@
-            point = gs.to_ndarray(tangent_vec, to_ndim=2)
-        elif point_type == 'matrix' and self.n > 3:
-            return gs.linalg.expm(tangent_vec)
-        elif point_type == 'matrix':
-            tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=3)
-            tangent_vec = self.vector_from_skew_matrix(tangent_vec)
-            point = self.matrix_from_rotation_vector(tangent_vec)
-
-        return point
-
-    def log_from_identity(self, point, point_type=None):
-        """Compute the group logarithm of the point at the identity.
-
-        Parameters
-        ----------
-        point : array-like, shape=[n_samples, {dimension, [n, n]}]
-        point_type : str, {'vector', 'matrix'}, optional
-            default: self.default_point_type
-
-        Returns
-        -------
-        tangent_vec : array-like, shape=[n_samples, {dimension, [n, n]}]
-        """
-        if point_type is None:
-            point_type = self.default_point_type
-
-        if point_type == 'vector':
-            tangent_vec = self.regularize(
-                point, point_type=point_type)
-        elif point_type == 'matrix':
-            point = self.rotation_vector_from_matrix(point)
-            tangent_vec = self.skew_matrix_from_vector(point)
-        return tangent_vec
-
-    def exponential_barycenter(
-            self, points, weights=None, point_type=None):
-        """Compute the group exponential barycenter in SO(n).
-
-        This is the Frechet mean of the canonical bi-invariant metric on SO(n).
-
-        Parameters
-        ----------
-        points : array-like, shape=[n_samples, {dimension, [n, n]}]
-        weights : array-like, shape=[n_samples], optional
-            default: 1 / n_samples for each point
-        point_type : str, {'vector', 'matrix'}, optional
-            default: self.default_point_type
-
-        Returns
-        -------
-        exp_bar : array-like, shape=[{dimension, [n, n]}]
-            the group exponential barycenter
-        """
-        if point_type is None:
-            point_type = self.default_point_type
-
-        if point_type == 'vector':
-            n_points = points.shape[0]
-            assert n_points > 0
-
-            if weights is None:
-                weights = gs.ones((n_points, 1))
-
-            n_weights = weights.shape[0]
-            assert n_points == n_weights
-
-            mean = FrechetMean(metric=self.bi_invariant_metric)
-            mean.fit(X=points, weights=weights)
-            exp_bar = mean.estimate_
-
-            exp_bar = gs.to_ndarray(exp_bar, to_ndim=2)
-            assert gs.ndim(exp_bar) == 2, gs.ndim(exp_bar)
-
-        elif point_type == 'matrix':
-            points = self.rotation_vector_from_matrix(points)
-            exp_bar = self.exponential_barycenter(
-                points, weights, point_type='vector')
-            exp_bar = self.matrix_from_rotation_vector(exp_bar)
-
-        return exp_bar
+        self.dim = int((n * (n - 1)) / 2)
```

Comment:
Yes to rename dimension into `dim`, see issue #423.

But let's keep this change to a PR that does only this change of naming and does it everywhere? Unless the code is already inconsistent?